swan-lake/learn-the-platform/ballerina-persist/persist-client-api.md

```diff
@@ -0,0 +1,245 @@
+    ```ballerina
+    public type WorkspaceOptionalized record {|
+        readonly string id?;
+        string workspaceType?;
+        string locationCode?;
+    |};
+    ```
+   
+5. With Relations Types
+    This types inherits all fields from the corresponding `Optionalized` type, and adds the relation fields as optional. This type is not directly used in any operations.
+    ```ballerina
+    public type WorkspaceWithRelations record {|
+        *WorkspaceOptionalized;
+        BuildingOptionalized location?;
+        EmployeeOptionalized employees?;
+    |};
+    ```
+
+6. Target Types
+    This type is used to retrieve data from the data source. If the entity contains a relation field, the target type 
+    will be a type description of the corresponding `WithRelations` type. Otherwise, the target type will be a 
+    type description of the corresponding `Optionalized` type.
+    ```ballerina
+    public type WorkspaceTargetType typedesc<WorkspaceWithRelations>;
+    ```
+    
+### Persist Clients
+
+Persist Clients are derived for each data model definition file. The client is used to perform CRUD operations on the data source. Each entity type will have five resource methods for each operation. The resource methods are get, get(get by identity), post, put, and delete.
+
+The skeleton of the Persist client is as follows,
+```ballerina
+public client class Client {
+    *persist:AbstractPersistClient;
+
+    public function init() returns persist:Error? {
+    }
+
+    isolated resource function get workspaces(WorkspaceTargetType targetType = <>) returns stream<targetType, Error?> {
+    };
+
+    isolated resource function get workspaces/[string id](WorkspaceTargetType targetType = <>) returns targetType|Error {
+    };
+
+    isolated resource function post workspaces(WorkspaceInsert[] data) returns string[]|persist:Error {
+    };
+
+    isolated resource function put workspaces/[string id](WorkspaceUpdate data) returns Workspace|persist:Error {
+    };
+
+    isolated resource function delete workspaces/[string id]() returns Workspace|persist:Error {
+    };
+
+    public function close() returns persist:Error? {
+    }
+}
+```
+
+The conventions used in deriving the Persist client are as follows:
+1. Since there can be only one generated client in a Ballerina package, the Client name is always `Client`.
+2. The client should be of the `persist:AbstractPersistClient` type.
+3. It should contain `init()` and `close()` functions.
+4. It should contain five resource methods for each entity type defined in the data model. Resource methods are get, get(get by identity), post, put, and delete.
+5. Resource names are plural form of lowercase entity names.
+6. The resource method should return the derived entity types.
+7. Resource method with path parameters will support composite identity field by having multiple path parameters.
+
+### Persist Client Usage
+
+#### Client Initialization
```

Comment:
```suggestion
#### Client initialization
```
All titles should be in sentence case. Please check all titles on all the pages.